res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -417,6 +417,16 @@ const mixerStrip = function(deckNumber, midiOffset) {
         });
     }
 
+    // VU Meters
+    this.vuMeter = new components.Component({
+        midi: [0xB0 + midiOffset, 0x0A],
+        group: "[Channel" + deckNumber + "]",
+        outKey: "VuMeter",
+        output: function(_channel, _control, _value, _status, _group) {
+            this.send(0x7f);
+        },
+    });
```

Comment:
:+1: thats the correct approach for adding output. Good job. For a proper implementation I suggest you just overwrite `outValueScale` to map from the `0...1` range to whatever range is appropriate for your controller.

Reply:
Thanks :) I used the Roland 505 mapping to help. The Prime 4's value range for the VU meters is a bit weird (see my [Prime 4 Analysis Repo](https://github.com/whanake-music/denon-prime-4-analysis) for more info on that) but I think I got it working pretty well for now.